apps/console/src/public/resources/users/init-impersonate.html

```diff
@@ -0,0 +1,62 @@
+<!--
+  ~ Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+  ~
+  ~ WSO2 LLC. licenses this file to you under the Apache License,
+  ~ Version 2.0 (the "License"); you may not use this file except
+  ~ in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~ http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+-->
+
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    </head>
+    <body onload="authenticate()">
+        <script>
+            var hash = window.location.hash;
+            function authenticate() {
+                if (!hash) {
+                    if (window.location.href.includes("oauth2_error.do")) {
+                        if (sessionStorage.getItem("impersonation_artifacts") === null) {
+                            sessionStorage.setItem("impersonation_artifacts", "oauth2_error");
+                        }
+                        window.parent.postMessage("impersonation-authorize-request-complete", "*");
+                    } else {
+                        const params = new URLSearchParams(window.location.search);
+                        const userId = params.get("userId");
+                        const code_challenge = params.get("codeChallenge");
+                        const client_id = params.get("clientId");
+                        const authorization_endpoint = sessionStorage.getItem("authorization_endpoint");
+                        window.location.href = authorization_endpoint
+                            + "?client_id=" + client_id +
+                            "&redirect_uri=" + window.location.origin + window.location.pathname +
+                            "&state=sample_state&scope=internal_user_impersonate&response_type=id_token%20subject_token" +
```

Comment:
is sending same state value intentional?

Reply:
Resolved with e566b44176dd034f329c9cfb9c4a4d39093c7f56